hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1186,6 +1264,72 @@ func (r *NodePoolReconciler) getConfig(ctx context.Context,
 	return strings.Join(allConfigPlainText, "\n---\n"), missingConfigs, utilerrors.NewAggregate(errors)
 }
 
+func (r *NodePoolReconciler) getTunedConfig(ctx context.Context,
+	nodePool *hyperv1.NodePool,
+) (configsRaw string, err error) {
+	var configs []corev1.ConfigMap
+	var allConfigPlainText []string
+	var errors []error
+
+	for _, config := range nodePool.Spec.TunedConfig {
+		configConfigMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      config.Name,
+				Namespace: nodePool.Namespace,
+			},
+		}
+		if err := r.Get(ctx, client.ObjectKeyFromObject(configConfigMap), configConfigMap); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		configs = append(configs, *configConfigMap)
+	}
+
+	for _, config := range configs {
+		manifestRaw := config.Data[tunedConfigKey]
+		manifest, err := defaultAndValidateTunedConfigManifest([]byte(manifestRaw))
+		if err != nil {
+			errors = append(errors, fmt.Errorf("configmap %q failed validation: %w", config.Name, err))
+			continue
+		}
+
+		allConfigPlainText = append(allConfigPlainText, string(manifest))
+	}
+
+	// Keep output deterministic to avoid unnecesary no-op changes to Tuned ConfigMap
+	sort.Strings(allConfigPlainText)
+	return strings.Join(allConfigPlainText, "\n---\n"), utilerrors.NewAggregate(errors)
+}
+
+func defaultAndValidateTunedConfigManifest(manifest []byte) ([]byte, error) {
+	scheme := runtime.NewScheme()
+	tunedv1.AddToScheme(scheme)
+
+	YamlSerializer := serializer.NewSerializerWithOptions(
+		serializer.DefaultMetaFactory, scheme, scheme,
+		serializer.SerializerOptions{Yaml: true, Pretty: true, Strict: true},
+	)
+
+	cr, _, err := YamlSerializer.Decode(manifest, nil, nil)
```

Comment:
How does defaulting happen with just decode/encode?

Reply:
I renamed this func, since at least for now, we aren't defaulting the objects.